tests/lib/bin/mountinfo-tool

```diff
@@ -0,0 +1,379 @@
+            # one based on cardinality of the set of major numbers we've seen.
+            major: int
+            for orig, remapped in seen.items():
+                if orig.major == dev.major:
+                    major = remapped.major
+                    break
+            else:
+                major = len({orig.major for orig in seen})
+            # Allocate the next minor number based on the cardinality of the
+            # set of minor numbers matching the major number.
+            minor = len({orig.minor for orig in seen
+                        if orig.major == dev.major})
+            n = Device.pack(major, minor)
+            seen[key] = n
+            return n
+    entry.dev = alloc_n(entry.dev)
+
+
+def rewrite_renumber(entries: List[MountInfoEntry]) -> None:
+    """rewrite_renumber applies all re-numbering helpers."""
+    seen_opt_fields: Dict[int, int] = {}
+    seen_loops: Dict[int, int] = {}
+    seen_snap_revs: Dict[Tuple[str, str], int] = {}
+    seen_mount_ids: Dict[int, int] = {}
+    seen_devices: Dict[Device, Device] = {}
+    for entry in entries:
+        renumber_mount_ids(entry, seen_mount_ids)
+        renumber_devices(entry, seen_devices)
+        renumber_snap_revision(entry, seen_snap_revs)
+        renumber_opt_fields(entry, seen_opt_fields)
+        renumber_loop_devices(entry, seen_loops)
+
+
+def rewrite_rename(entries: List[MountInfoEntry]) -> None:
+    """rewrite_renameapplies all re-naming helpers."""
+    # TODO: allocate devices like everything else above.
+    for entry in entries:
+        entry.mount_source = re.sub(
+            '/dev/[sv]d([a-z])', '/dev/sd\\1', entry.mount_source)
+
+
+def main() -> None:
+    parser = ArgumentParser(epilog="""
+Expressions are ANDed together and have one of the following forms:
+
+    .ATTR=VALUE     mount entry attribute ATTR is equal to VALUE
+    PATH            mount point is equal to PATH
+    PATH...         mount point starts with PATH
+
+In addition .ATTR syntax can be used to limit display to only certain
+attributes. By default the output is identical to raw mountinfo.
+Known attributes, applicable for both filtering and display.
+
+    mount_point:    path where mount is attached in the file system
+    mount_source:   path of the mounted device or bind-mount origin
+    fs_type:        filesystem type
+    mount_opts:     options applying to the mount point only
+    sb_opts:        options applying to the mounted filesystem
+    opt_fields:     optional fields, used for propagation information
+    mount_id:       mount point identifier
+    parent_id:      identifier of parent mount point
+    dev:            major:minor numbers of the mounted device
+    root_dir:       subtree of the mounted filesystem exposed at mount_point
+    """, formatter_class=RawTextHelpFormatter)
+    parser.add_argument('-v', '--version', action="version", version="1.0")
+    parser.add_argument(
```

Comment:
I tested that with just -f and it is not using the default value:
> ./mountinfo-tool -f
usage: mountinfo-tool [-h] [-v] [-f MOUNTINFO] [--one] [--renumber] [--rename]
                      [EXPRESSION [EXPRESSION ...]]
mountinfo-tool: error: argument -f: expected one argument

Reply:
The `-f` option accepts a file name to use instead of `/proc/self/mountinfo`, normally you can just skip it. Alternatively you can use `-f-` to parse stdin.